core/src/main/java/org/jdbi/v3/core/async/JdbiExecutor.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.async;
+
+import java.util.concurrent.CompletionStage;
+
+import org.jdbi.v3.core.HandleCallback;
+import org.jdbi.v3.core.HandleConsumer;
+import org.jdbi.v3.core.extension.ExtensionCallback;
+import org.jdbi.v3.core.extension.ExtensionConsumer;
+import org.jdbi.v3.core.extension.ExtensionFactory;
+import org.jdbi.v3.core.transaction.TransactionIsolationLevel;
+
+public interface JdbiExecutor {
+
+    /**
+     * A convenience function which manages the lifecycle of a handle and yields it to a callback
+     * for use by clients.
+     *
+     * <p>
+     * The callback will be executed in a separate thread
```

Comment:
Strictly speaking, this is not necessarily true. For example Guava has a "direct" executor which runs on the current thread, and there is the `ThreadPoolExecutor.CallerRunsPolicy` which does the same.

Reply:
Fair point. Changed it to say "will be executed in a thread supplied by the executor". That should cover these cases